src/main/java/com/google/sps/Trip.java

```diff
@@ -0,0 +1,190 @@
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Date;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+
+/**
+ * Trip is the class for storing a single trip (could be multiple days).
+ */
+public class Trip {
+  private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd");
+
+  private String tripName;
+  private String startDate;
+  private String endDate;
+  private int numDays;
+  private List<TripDay> tripDays;
+
+  /**
+   * Creates a new Trip.
+   *
+   * @param tripName The human-readable name for the trip. Must be non-null.
+   * @param startDate The start date for the trip. Must be non-null.
+   * @param endDate The end date for the trip. Must be non-null.
+   * @param numDays The number of days for the trip. 
+   * @param tripDays The list of tripDays. Must be non-null.
+   */
+  public Trip(String tripName, String startDate, String endDate, List<TripDay> tripDays) {
+    if (tripName == null) {
+      throw new IllegalArgumentException("tripName cannot be null");
+    }
+
+    if (startDate == null) {
+      throw new IllegalArgumentException("startDate cannot be null");
+    }
+
+    if (endDate == null) {
+      throw new IllegalArgumentException("endDate cannot be null");
+    }
+
+    if (tripDays == null) {
+      throw new IllegalArgumentException("tripDays cannot be null. Use empty array instead.");
+    }
+
+    if (!isValidDate(startDate)) {
+      throw new IllegalArgumentException("Invalid startDate format.");
+    }
+
+    if (!isValidDate(endDate)) {
+      throw new IllegalArgumentException("Invalid endDate format.");
+    }
+
+    int numDays = calcNumDays(startDate, endDate);
+
+    if (numDays < 0 || numDays > 31) {
+      throw new IllegalArgumentException("numDays must be an integer between 1 and 31.");
+    }
+
+    this.tripName = tripName;
+    this.startDate = startDate;
+    this.endDate = endDate;
+    this.numDays = numDays;
+    
+    // Duplicate tripDays to avoid modifying original parameter
+    this.tripDays = new ArrayList<>();
+    this.tripDays.addAll(tripDays);
+  }
+
+  /**
+   * Creates a new Trip that is only one day by default. 
+   * This constructor will be used for the MVP.
+   * 
+   * @param tripName The human-readable name for the trip. Must be non-null.
+   * @param startDate The start date for the trip. Must be non-null.
+   * @param tripDays The list of tripDays. Must be non-null.
+   */
+  public Trip(String tripName, String startDate, List<TripDay> tripDays) {
```

Comment:
I believe you can call your other constructor from here, which will reduce a lot of the code duplication.

public Trip(String tripName, String startDate, List<TripDay> tripDays) {
  this(tripName, startDate, startDate, tripDays);
}